src/libcore/iter.rs

```diff
@@ -2224,13 +2222,11 @@ impl<I> DoubleEndedIterator for Fuse<I> where I: DoubleEndedIterator {
         if self.done {
             None
         } else {
-            match self.iter.next_back() {
-                None => {
-                    self.done = true;
-                    None
-                }
-                x => x
+            let next = self.iter.next_back();
+            if next.is_none() {
+                self.done = true;
```

Comment:
I was about to say the same thing as @lfairy , but I too don't feel too strongly about it, especially considering how small of a change this commit is